storage/zynq7000-flash/flashdrv.c

```diff
@@ -20,26 +20,33 @@
 #include <string.h>
 #include <sys/time.h>
 #include <sys/threads.h>
+#include <cache.h>
 
 /* Value determined empirically, contains time for communication via qspi and system calls invocations. */
 #define TIMEOUT_CMD_MS 1000
 #define MAX_SIZE_CMD   0x20
 
 #define MTD_DEFAULT_ERASESZ 0x10000
 
+#define BLK_CACHE_SECNUM 4 /* Maximum number of cached sectors in a region */
 
-typedef struct {
-	/* Data caching in sector's buffer */
-	uint8_t *buff;
-	uint32_t buffPos;
-	uint32_t sectID;
 
+/* Cached device context definition */
+struct cache_devCtx_s {
+	off_t start;     /* storage start */
+	unsigned int id; /* flash device memory id */
+};
+
+
+typedef struct {
 	/* Buffers for command transactions */
 	uint8_t cmdRx[MAX_SIZE_CMD];
 	uint8_t cmdTx[MAX_SIZE_CMD];
 
 	handle_t lock;
 	off_t start;
+
+	cachectx_t *cache;
```

Comment:
So there's one less malloc
```suggestion
	cachectx_t cache;
```